datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -465,7 +465,116 @@ impl FileFormat for ParquetFormat {
     }
 }
 
+/// Apply necessary schema type coercions to make file schema match table schema.
+///
+/// This function performs two main types of transformations in a single pass:
+/// 1. Binary types to string types conversion - Converts binary data types to their
+///    corresponding string types when the table schema expects string data
+/// 2. Regular to view types conversion - Converts standard string/binary types to
+///    view types when the table schema uses view types
+///
+/// # Arguments
+/// * `table_schema` - The table schema containing the desired types
+/// * `file_schema` - The file schema to be transformed
+///
+/// # Returns
+/// * `Some(Schema)` - If any transformations were applied, returns the transformed schema
+/// * `None` - If no transformations were needed
+pub fn apply_file_schema_type_coercions(
+    table_schema: &Schema,
+    file_schema: &Schema,
+) -> Option<Schema> {
+    let mut needs_view_transform = false;
+    let mut needs_string_transform = false;
+
+    // Create a mapping of table field names to their data types for fast lookup
+    // and simultaneously check if we need any transformations
+    let table_fields: HashMap<_, _> = table_schema
+        .fields()
+        .iter()
+        .map(|f| {
+            let dt = f.data_type();
+            // Check if we need view type transformation
+            if dt.equals_datatype(&DataType::Utf8View)
```

Comment:
`equals_datatype` is used for nested type, in this case `matches!()` is enough

Reply:
fixed in [resolve comments](https://github.com/apache/datafusion/pull/15268/commits/36ae7c7b8af641184b8b811e9b7a513ab5243149)